src/external_trait_impls/serde.rs

```diff
@@ -0,0 +1,116 @@
+use core::fmt;
+use core::marker::PhantomData;
+
+use serde_::de::{Deserialize, Deserializer, SeqAccess, Visitor};
+use serde_::ser::{Serialize, Serializer};
+
+use crate::Vc;
+
+impl<T> Serialize for Vc<T>
+where
+    T: Serialize,
+{
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        serializer.collect_seq(self)
+    }
+}
+
+#[inline]
+fn cautious_size_hint(hint: Option<usize>) -> usize {
+    core::cmp::min(hint.unwrap_or(0), 4096)
```

Comment:
Let's add a comment here pointing to https://github.com/serde-rs/serde/blob/9f331cc25753edd71ad7ab0ea08a430fefaa90e1/serde/src/private/de.rs#L203 to show that we didn't pull this out of a hat.